private/buf/cmd/buf/command/push/push.go

```diff
@@ -347,6 +394,415 @@ func validateLabelFlagValues(flags *flags) error {
+	if err != nil {
+		return nil, err
+	}
+	if gitLabelsUploadOption != nil {
+		gitMetadataUploadOptions = append(gitMetadataUploadOptions, gitLabelsUploadOption)
+	}
+	gitSourceControlURLUploadOption, err := getGitMetadataSourceControlURLUploadOption(ctx, runner, container.Logger(), remotes, input)
+	if err != nil {
+		return nil, err
+	}
+	if gitSourceControlURLUploadOption != nil {
+		gitMetadataUploadOptions = append(gitMetadataUploadOptions, gitSourceControlURLUploadOption)
+	}
+	if flags.Create {
+		gitDefaultBranch, err := getGitDefaultBranch(ctx, runner, container, remotes, input)
+		if err != nil {
+			return nil, err
+		}
+		createModuleVisibility, err := bufmodule.ParseModuleVisibility(flags.CreateVisibility)
+		if err != nil {
+			return nil, err
+		}
+		gitMetadataUploadOptions = append(
+			gitMetadataUploadOptions,
+			bufmodule.UploadWithCreateIfNotExist(createModuleVisibility, gitDefaultBranch),
+		)
+	}
+	return gitMetadataUploadOptions, nil
+}
+
+func checkForUncommittedGitChanges(
+	ctx context.Context,
+	runner command.Runner,
+	input string,
+) ([]string, error) {
+	stdout := bytes.NewBuffer(nil)
+	stderr := bytes.NewBuffer(nil)
+	var modifiedFiles []string
+	// Unstaged changes
+	if err := runner.Run(
+		ctx,
+		gitCommand,
+		command.RunWithArgs("diff", "--name-only"),
+		command.RunWithStdout(stdout),
+		command.RunWithStderr(stderr),
+		command.RunWithDir(input),
+	); err != nil {
+		return nil, err
+	}
+	modifiedFiles = append(modifiedFiles, getAllTrimmedLinesFromBuffer(stdout)...)
+
+	stdout = bytes.NewBuffer(nil)
+	stderr = bytes.NewBuffer(nil)
+	// Staged changes
+	if err := runner.Run(
+		ctx,
+		gitCommand,
+		command.RunWithArgs("diff", "--name-only", "--cached"),
+		command.RunWithStdout(stdout),
+		command.RunWithStderr(stderr),
+		command.RunWithDir(input),
+	); err != nil {
+		return nil, err
+	}
+
+	modifiedFiles = append(modifiedFiles, getAllTrimmedLinesFromBuffer(stdout)...)
+	return modifiedFiles, nil
+}
+
+func getGitRemotes(
+	ctx context.Context,
+	runner command.Runner,
+	logger *zap.Logger,
+	input string,
+) ([]string, error) {
+	buffer := bytes.NewBuffer(nil)
+	if err := runner.Run(
+		ctx,
+		gitCommand,
+		command.RunWithArgs("remote"),
+		command.RunWithStdout(buffer),
+		command.RunWithStderr(buffer),
+		command.RunWithDir(input),
+	); err != nil {
+		return nil, err
+	}
+	scanner := bufio.NewScanner(buffer)
+	var gitOriginRemoteFound bool
+	var remotes []string
+	for scanner.Scan() {
+		remote := strings.TrimSpace(scanner.Text())
+		if remote == gitOriginRemote {
+			gitOriginRemoteFound = true
+		}
+		remotes = append(remotes, remote)
+	}
+	if !gitOriginRemoteFound {
+		return nil, fmt.Errorf("Git remote %q not found", gitOriginRemote)
+	}
+	if len(remotes) > 1 {
+		logger.Warn(
+			fmt.Sprintf("More than one Git remote found, %q will be used for Git metadata: %v", gitOriginRemote, remotes),
+		)
+	}
```

Comment:
So the decision was that we wouldn't fail if there is more than one remote (basically exactly because of the forked upstreams situation -- you may get additional remotes from forks), but we do fail if the remote we care about, `origin` isn't present, and we basically only care about `origin`.

I think if we are sticking to this, then the only relevant remote is only ever `origin`, so we may not even need to return any here and simply use this to validate remotes instead.

Reply:
> may not even need to return any here and simply use this to validate remotes instead

Did you continue to return all remotes in the event that we revert this decision? If so, maybe a comment to that effect would be helpful -- a number of my comments were about this.

Reply:
It wasn't so much in case we revert so much as things were a bit in-flight as code was being written. I think if that's the UX we're going for, it's totally fine to have the code more closely reflect that :) So i'm onboard with making these changes.